### PR TITLE
[VL] Add bin function support

### PR DESCRIPTION
**Pull request by Yohahaha**

### CODE REVIEW

**Comment by Yohahaha** [comment]
@rui-mo @PHILO-HE @zhejiangxiaomai could you help take a look?

**Comment by PHILO-HE** [comment]
Looks there is no need to do such change: https://github.com/oap-project/velox/pull/349/files. That is a white list for agg function.

**Comment by Yohahaha** [comment]
> Thanks for your proposed patch! Looks there is no need to do such change: https://github.com/oap-project/velox/pull/349/files. That is a white list for agg function. Right?

Changs in oap/velox is needed, more details about that could ask @rui-mo

File: backends-velox/src/test/scala/io/glutenproject/execution/TestOperator.scala [hunk 1]
```diff
@@ -243,6 +243,12 @@ class TestOperator extends WholeStageTransformerSuite {
     checkLengthAndPlan(df, 1)
   }
 
+  test("bin function") {
+    val df = runQueryAndCompare("SELECT bin(l_orderkey) " +
+      "from lineitem limit 1") { _ => }
+    checkLengthAndPlan(df, 1)
```
Comment (rui-mo) [inline backends-velox/src/test/scala/io/glutenproject/execution/TestOperator.scala:249]: Can we use `checkOperatorMatch` to assert a Project operator?

Comment (Yohahaha) [reply backends-velox/src/test/scala/io/glutenproject/execution/TestOperator.scala:251]: Already updated, thanks!

**Comment by PHILO-HE** [comment]
Some warning fixes are covered by https://github.com/oap-project/gluten/pull/2212. Please do a rabase after that PR is merged. Thanks!

**Review by PHILO-HE — APPROVED** [review]
Looks great! Thanks!

**Pull request merged**